### PR TITLE
Feature/adrs

### PR DESCRIPTION
**Pull request by st3v3nhunt**

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployment of connecting-to-services succeeded (http://connecting-to-services-pr-296.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of connecting-to-services succeeded (http://connecting-to-services-pr-296.dev.beta.nhschoices.net)

File: doc/architecture/decisions/0002-use-express-web-framework.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+such has been well tested and is deployed across many
+[organisations](http://expressjs.com/en/resources/companies-using-express.html).
+
+## Consequences
+
+Due to Express being a minimalist web framework, when additional features are
+required they are added as
+[middleware](http://expressjs.com/en/resources/middleware.html) modules.
+A working knowledge of existing modules will help with reducing duplication of
+effort by where modules may already exist within the community.
```
Comment (markysoft) [inline doc/architecture/decisions/0002-use-express-web-framework.md]: This sentence `A working knowledge of existing modules will help with reducing duplication of effort by where modules may already exist within the community` doesn't scan too well.
Could it be shortened to something like `Duplication of effort for common tasks may be reduced by using modules that already exist within the community`

File: doc/architecture/decisions/0003-use-postcodes-io-for-postcode-lookup.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+[ONS](https://data.gov.uk/dataset/national-statistics-postcode-lookup-uk).
+
+## Consequences
+
+Only currently active postcodes return responses. This means any
+deactivated/deleted postcodes return 404s from which a co-ordinate can not be
+retrieved and the application will not be able to return any results.
+If additional search terms e.g. town names are added to the application an
+additional lookup service might be required. That service might be instead of
+Postcodes.io or as well was.
```
Comment (markysoft) [inline doc/architecture/decisions/0003-use-postcodes-io-for-postcode-lookup.md]: typo: 'was' should be 'as'. Alternatively change to `That service might replace or be used alongside Postcodes.io`.

File: doc/architecture/decisions/0004-use-api-layer-for-returning-services.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+## Context
+
+The service data had been stored in a flat file and with the use of a number of
+libraries [geolib](https://www.npmjs.com/package/geolib),
+[ngeohash](https://www.npmjs.com/package/ngeohash) and
+[geo-nearby](https://www.npmjs.com/package/geo-nearby) the capabilities of a
+geo-location search was added. However, this is a CPU intense operation. Node
+is not best suited to that use case. This resulted in the Node process being
+blocked for the duration of the calculation. The performance of the site wasn't
+effected too much when services (that were open) were found close to the search
```
Comment (markysoft) [inline doc/architecture/decisions/0004-use-api-layer-for-returning-services.md]: should be `affected` rather than `effected`

**Comment by st3v3nhunt** [comment]
@markysoft Good spots/suggestions. Changes have been made.

**Comment by c2s-dev** [comment]
:rocket: deployment of connecting-to-services succeeded (http://connecting-to-services-pr-296.dev.beta.nhschoices.net)

**Pull request merged**